aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java

```diff
@@ -67,6 +77,212 @@ public ApiInfoDto getAipConsoleApiInfo() {
+            }
+
+            applicationGuid = app.getGuid();
+            ApplicationOnboardingDto applicationOnboardingDto = getApplicationOnboarding(applicationGuid);
+            String caipVersion = applicationOnboardingDto.getCaipVersion();
+            String targetNode = applicationOnboardingDto.getTargetNode();
+
+            DeliveryConfigurationDto deliveryConfiguration = null;
+            Exclusions exclusions = Exclusions.builder().excludePatterns(fastScanProperties.getExclusionPatterns()).build();
+            if (fastScanProperties.getExclusionRules() != null && fastScanProperties.getExclusionRules().length > 0) {
+                exclusions.setInitialExclusionRules(fastScanProperties.getExclusionRules());
+            }
+
+            //discover-packages
+            log.info("Preparing the Application Delivery Configuration");
+            DeliveryConfigurationDto[] deliveryConfig = new DeliveryConfigurationDto[1];
+            String deliveryConfigurationGuid = discoverPackagesAndCreateDeliveryConfiguration(applicationGuid, sourcePath, exclusions,
+                    VersionStatus.DELIVERED, true, (config) -> deliveryConfig[0] = config, true);
+            deliveryConfiguration = deliveryConfig[0];
+            log.info("Application Delivery Configuration done: GUID=" + deliveryConfigurationGuid);
+            deliveryConfiguration.setGuid(deliveryConfigurationGuid);
+
+            log.info("Starting Fast-Scan action with Delivery Configuration Guid=" + deliveryConfiguration.getGuid());
+            String jobStatus = fastScan(applicationGuid, sourcePath, "", deliveryConfiguration,
+                    caipVersion, targetNode, fastScanProperties.isVerbose(), fastScanProperties.getLogPollingProvider());
+            if (jobStatus != null && jobStatus.equalsIgnoreCase(JobState.COMPLETED.toString())) {
+                log.info("Fast-Scan done successfully");
+            } else {
+                log.info("Fast-Scan not completed successfully");
+                return Constants.RETURN_JOB_FAILED;
+            }
+        } catch (ApplicationServiceException e) {
+            return Constants.RETURN_APPLICATION_INFO_MISSING;
+        }
+
+        return Constants.RETURN_OK;
+    }
+
+    @Override
+    public int deepAnalyze(DeepAnalyzeProperties properties) throws JobServiceException {
+        log.info("Deep-Analyze command will perform application '{}' deep analysis: verbose= '{}'"
+                , properties.getApplicationName(), properties.isVerbose());
+        log.info("Deep-Analysis args:");
+        log.info(String.format("\tApplication: %s%n\tsnapshot name: %s%n\tmodule generation type: %s%n\tsleep: %d%n"
+                , properties.getApplicationName(), StringUtils.isEmpty(properties.getSnapshotName()) ? "Auto assigned" : properties.getSnapshotName()
+                , properties.getModuleGenerationType().toString(), properties.getSleepDuration()));
+
```

Comment:
surround with a `if(log.isInfoEnabled())`, if you need to have a large amount of text generated to send to the log, but the info level is not enabled, you're wasting time constructing the messages that will be discarded :/

Also, if you are using SLF4J, you shouldn't use String.format. You can directly use "{}" inside you log message and pass that as a parameter to the method; for example : 

```java
log.info("Application: {}\nSnapshot name: {}\nModule Generation Type: {}\nSleep: {}",
        properties.getApplicationName(),
        StringUtils.defaultIfEmpty(properties.getSnapshotName(), "Auto assigned"),
        properties.getModuleGenerationType(),
        properties.getSleepDuration());
```
`log.info` will handle calling "toString" when needed, so you don't have to do it yourself. I've also used the StringUtils.defaultifEmpty which will do exactly what you are doing, but is a bit more readable

Reply:
I am using Slf4j